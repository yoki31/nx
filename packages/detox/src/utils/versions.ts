export const nxVersion = require('../../package.json').version;

export const detoxVersion = '~20.18.1';
export const testingLibraryJestDom = '~6.4.2';
export const configPluginsDetoxVersion = '~8.0.0'; // only required for expo
