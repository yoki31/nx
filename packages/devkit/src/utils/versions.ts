export const typescriptVersion = '~5.4.2';
