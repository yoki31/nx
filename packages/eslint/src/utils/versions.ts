export const nxVersion = require('../../package.json').version;

export const eslintVersion = '~8.57.0';
export const eslintrcVersion = '^2.1.1';
export const eslintConfigPrettierVersion = '^9.0.0';
export const typescriptESLintVersion = '^7.3.0';
