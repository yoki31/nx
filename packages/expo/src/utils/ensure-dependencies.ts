import {
  addDependenciesToPackageJson,
  detectPackageManager,
  type GeneratorCallback,
  type Tree,
} from '@nx/devkit';
import {
  babelPresetExpoVersion,
  babelRuntimeVersion,
  expoMetroConfigVersion,
  expoMetroRuntimeVersion,
  expoSplashScreenVersion,
  expoStatusBarVersion,
  jestExpoVersion,
  reactNativeSvgTransformerVersion,
  reactNativeSvgVersion,
  reactNativeWebVersion,
  reactTestRendererVersion,
  testingLibraryJestNativeVersion,
  testingLibraryReactNativeVersion,
  typesReactVersion,
} from './versions';

export function ensureDependencies(host: Tree): GeneratorCallback {
  const isPnpm = detectPackageManager(host.root) === 'pnpm';
  return addDependenciesToPackageJson(
    host,
    {
      'expo-splash-screen': expoSplashScreenVersion,
      'expo-status-bar': expoStatusBarVersion,
      'react-native-web': reactNativeWebVersion,
      '@expo/metro-config': expoMetroConfigVersion,
      '@expo/metro-runtime': expoMetroRuntimeVersion,
      'react-native-svg-transformer': reactNativeSvgTransformerVersion,
      'react-native-svg': reactNativeSvgVersion,
    },
    {
      '@types/react': typesReactVersion,
      'react-test-renderer': reactTestRendererVersion,
      '@testing-library/react-native': testingLibraryReactNativeVersion,
      '@testing-library/jest-native': testingLibraryJestNativeVersion,
      'jest-expo': jestExpoVersion,
      'babel-preset-expo': babelPresetExpoVersion,
      ...(isPnpm
        ? {
            '@babel/runtime': babelRuntimeVersion, // @babel/runtime is used by react-native-svg
          }
        : {}),
    }
  );
}
