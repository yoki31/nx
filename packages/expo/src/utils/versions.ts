export const nxVersion = require('../../package.json').version;

export const expoVersion = '~51.0.8';
export const expoSplashScreenVersion = '~0.27.4';
export const expoStatusBarVersion = '~1.12.1';
export const expoCliVersion = '~0.18.13'; // @expo/cli
export const babelPresetExpoVersion = '~11.0.0';

export const reactVersion = '18.2.0';
export const reactDomVersion = '18.2.0';
export const reactTestRendererVersion = '18.2.0';
export const typesReactVersion = '~18.2.45';

export const reactNativeVersion = '0.74.1';
export const reactNativeWebVersion = '~0.19.11';

export const expoMetroConfigVersion = '~0.18.1';
export const expoMetroRuntimeVersion = '~3.2.1';

export const reactNativeSvgTransformerVersion = '1.3.0';
export const reactNativeSvgVersion = '15.2.0';

export const testingLibraryReactNativeVersion = '~12.5.0';
export const testingLibraryJestNativeVersion = '~5.4.3';
export const jestExpoVersion = '~51.0.2';

export const babelRuntimeVersion = '7.24.5';
