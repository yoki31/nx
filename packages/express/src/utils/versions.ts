export const nxVersion = require('../../package.json').version;

export const expressVersion = '^4.18.1';
export const expressTypingsVersion = '4.17.13';
