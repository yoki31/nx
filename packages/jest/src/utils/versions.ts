export const nxVersion = require('../../package.json').version;
export const jestVersion = '^29.4.1';
export const babelJestVersion = '^29.4.1';
export const jestTypesVersion = '^29.4.0';
export const tsJestVersion = '^29.1.0';
export const tslibVersion = '^2.3.0';
export const swcJestVersion = '~0.2.36';
export const typesNodeVersion = '18.16.9';
export const tsNodeVersion = '10.9.1';
