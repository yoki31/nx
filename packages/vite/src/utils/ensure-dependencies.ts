import {
  addDependenciesToPackageJson,
  logger,
  type GeneratorCallback,
  type Tree,
} from '@nx/devkit';
import {
  edgeRuntimeVmVersion,
  happyDomVersion,
  jsdomVersion,
  vitePluginDtsVersion,
  vitePluginReactSwcVersion,
  vitePluginReactVersion,
} from './versions';

export type EnsureDependenciesOptions = {
  uiFramework: 'react' | 'none';
  compiler?: 'babel' | 'swc';
  includeLib?: boolean;
  testEnvironment?: 'node' | 'jsdom' | 'happy-dom' | 'edge-runtime' | string;
};

export function ensureDependencies(
  host: Tree,
  schema: EnsureDependenciesOptions
): GeneratorCallback {
  const devDependencies: Record<string, string> = {};

  if (schema.testEnvironment === 'jsdom') {
    devDependencies['jsdom'] = jsdomVersion;
  } else if (schema.testEnvironment === 'happy-dom') {
    devDependencies['happy-dom'] = happyDomVersion;
  } else if (schema.testEnvironment === 'edge-runtime') {
    devDependencies['@edge-runtime/vm'] = edgeRuntimeVmVersion;
  } else if (schema.testEnvironment !== 'node' && schema.testEnvironment) {
    logger.info(
      `A custom environment was provided: ${schema.testEnvironment}. You need to install it manually.`
    );
  }

  if (schema.uiFramework === 'react') {
    if (schema.compiler === 'swc') {
      devDependencies['@vitejs/plugin-react-swc'] = vitePluginReactSwcVersion;
    } else {
      devDependencies['@vitejs/plugin-react'] = vitePluginReactVersion;
    }
  }

  if (schema.includeLib) {
    devDependencies['vite-plugin-dts'] = vitePluginDtsVersion;
  }

  return addDependenciesToPackageJson(host, {}, devDependencies);
}
