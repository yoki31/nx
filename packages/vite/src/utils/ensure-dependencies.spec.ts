import { addDependenciesToPackageJson, readJson, type Tree } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { ensureDependencies } from './ensure-dependencies';
import { nxVersion } from './versions';

describe('@nx/vite:init', () => {
  let tree: Tree;

  beforeEach(() => {
    tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
  });

  it('should add vite packages and react-related dependencies for vite', () => {
    const existing = 'existing';
    const existingVersion = '1.0.0';
    addDependenciesToPackageJson(
      tree,
      { '@nx/vite': nxVersion, [existing]: existingVersion },
      { [existing]: existingVersion }
    );
    ensureDependencies(tree, {
      uiFramework: 'react',
    });
    const packageJson = readJson(tree, 'package.json');

    expect(packageJson).toMatchSnapshot();
  });

  it('should support --testEnvironment=jsdom', () => {
    ensureDependencies(tree, {
      testEnvironment: 'jsdom',
      uiFramework: 'none',
    });

    const packageJson = readJson(tree, 'package.json');

    expect(packageJson).toMatchSnapshot();
  });

  it('should support --testEnvironment=happy-dom', () => {
    ensureDependencies(tree, {
      testEnvironment: 'happy-dom',
      uiFramework: 'none',
    });

    const packageJson = readJson(tree, 'package.json');

    expect(packageJson).toMatchSnapshot();
  });

  it('should support --testEnvironment=edge-runtime', () => {
    ensureDependencies(tree, {
      testEnvironment: 'edge-runtime',
      uiFramework: 'none',
    });

    const packageJson = readJson(tree, 'package.json');

    expect(packageJson).toMatchSnapshot();
  });
});
