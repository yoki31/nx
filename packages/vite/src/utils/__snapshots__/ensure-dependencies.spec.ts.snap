// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@nx/vite:init should add vite packages and react-related dependencies for vite 1`] = `
{
  "dependencies": {
    "@nx/vite": "0.0.1",
    "existing": "1.0.0",
  },
  "devDependencies": {
    "@vitejs/plugin-react": "^4.2.0",
    "existing": "1.0.0",
  },
  "name": "@proj/source",
}
`;

exports[`@nx/vite:init should support --testEnvironment=edge-runtime 1`] = `
{
  "dependencies": {},
  "devDependencies": {
    "@edge-runtime/vm": "~3.0.2",
  },
  "name": "@proj/source",
}
`;

exports[`@nx/vite:init should support --testEnvironment=happy-dom 1`] = `
{
  "dependencies": {},
  "devDependencies": {
    "happy-dom": "~9.20.3",
  },
  "name": "@proj/source",
}
`;

exports[`@nx/vite:init should support --testEnvironment=jsdom 1`] = `
{
  "dependencies": {},
  "devDependencies": {
    "jsdom": "~22.1.0",
  },
  "name": "@proj/source",
}
`;
